core/src/main/java/org/jivesoftware/spark/ui/login/CertificatesManagerSettingsPanel.java

```diff
@@ -152,7 +157,33 @@ private void showCertificate() {
 		CertificateDialog certDialog = new CertificateDialog(localPreferences,
 				certControll.getCertificates().get(certTable.getSelectedRow()));
 	}
-
+	
+	private void addCertificate(){
+		
+		fileChooser.setAcceptAllFileFilterUsed(false);
+		fileChooser.addChoosableFileFilter(certFilter);
+		fileChooser.setFileFilter(certFilter);
+
+		int retVal = fileChooser.showOpenDialog(this);
+		if (retVal == JFileChooser.APPROVE_OPTION) {
+
+			File file = fileChooser.getSelectedFile();
+			
+			try {
+				fileField.setText(file.getAbsolutePath());
+				String certAlias = (String) JOptionPane.showInputDialog(null, Res.getString("dialog.provide.alias"));
+				if (certAlias == null) {
+					throw new IllegalArgumentException();
+				}
+				certControll.addCertificateToKeystore(file, certAlias);
+			} catch (KeyStoreException | CertificateException | NoSuchAlgorithmException | IOException ex) {
+				Log.error("Cannot upload certificate file", ex);
+			} catch (IllegalArgumentException ex) {
```

Comment:
An IllegalArgumentException is a runtime exception - which is an exception that is intended to be used when something occurs that is considered to be a programming error (not user in put error). You should not use these for things like User-input validation. For that, use a checked exception instead.

Runtime exceptions are generally not 'catched', but bubble their way up all the way to a very core class, which generically processes them in a big 'catch-all' block.

For some background, read https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html